### PR TITLE
Preventing CORB + some explanatory comments

### PR DESCRIPTION
**Pull request by lazyweirdo**
We let gateways guess mime types to be sent when requesting for scripts files

feel free to modify it
test it https://pulsartronic.com/public-gateway-checker/

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -35,24 +38,34 @@ checker.results.failed = function(node) {
...
 Status.prototype.check = function() {
-	const gatewayAndScriptHash = this.parent.gateway.replace(":hash", SCRIPT_HASH);
-	const rnd = encodeURIComponent(Math.random());
-	const src = `${gatewayAndScriptHash}?i=${this.parent.index}&rnd=${rnd}#x-ipfs-companion-no-redirect`;
-	let script = document.createElement('script')
+	let gatewayAndScriptHash = this.parent.gateway.replace(":hash", SCRIPT_HASH);
+
+	// we set a random number as a url parameter, to try to prevent content caching
+	// is it right ? ... do you know a better way ? ... does it always work ?
+	let rnd = Math.random();
```
Comment (lidel) [inline app.js]: Unix timestamp is bit better as it is guaranteed to be unique across page loads:
```suggestion
	let rnd = new Date()).getTime();
```

**Comment by lazyweirdo** [comment]
> @lazyweirdo do you got an example of gateway which was hit by CORB?

yep
https://10.via0.com
https://ipfs.eternum.io
https://ipfs.fooock.com
https://permaweb.io

File: app.js [hunk 2]
```diff
@@ -35,24 +38,34 @@ checker.results.failed = function(node) {
...
-	let script = document.createElement('script')
+	let gatewayAndScriptHash = this.parent.gateway.replace(":hash", SCRIPT_HASH);
+
+	// we set a random number as a url parameter, to try to prevent content caching
+	// is it right ? ... do you know a better way ? ... does it always work ?
+	let rnd = new Date()).getTime();
+
+	// 3 important things here
+	//   1) we add #x-ipfs-companion-no-redirect to the final url (self explanatory)
+	//   2) we add filename=anyname.js as a parameter to let the gateway guess a mime type
```
Comment (lidel) [inline app.js]:
```suggestion
	//   2) we add ?filename=anyname.js as a parameter to let the gateway set correct Content-Type header
```

File: app.js [hunk 3]
```diff
@@ -35,24 +38,34 @@ checker.results.failed = function(node) {
...
 Status.prototype.check = function() {
-	const gatewayAndScriptHash = this.parent.gateway.replace(":hash", SCRIPT_HASH);
-	const rnd = encodeURIComponent(Math.random());
-	const src = `${gatewayAndScriptHash}?i=${this.parent.index}&rnd=${rnd}#x-ipfs-companion-no-redirect`;
-	let script = document.createElement('script')
+	let gatewayAndScriptHash = this.parent.gateway.replace(":hash", SCRIPT_HASH);
+
+	// we set a random number as a url parameter, to try to prevent content caching
+	// is it right ? ... do you know a better way ? ... does it always work ?
+	let rnd = new Date()).getTime();
```
Comment (lidel) [inline app.js]: I made a typo before :( this needs below fix:
```suggestion
	let rnd = new Date().getTime();
```

**Comment by lazyweirdo** [comment]
> Ah, I get it. CORB was breaking the ping check:

yes, fortunately those gateways with CORB enabled have also CORS enabled, which makes them appear, at least, UP ....

**Comment by jamiew** [comment]
@lazyweirdo Lidel requested some changes which appear to still be outstanding, did you want to address those and we can merge in?

**Comment by lazyweirdo** [comment]
do i have to do further things here ??? it isn't clear ... why there isn't a giant button saying ... IM DONE, GO ON ... sorry about my ignorance, years working with svn ...

**Comment by jamiew** [comment]
Ah, hm, did you delete the fork you were merging from? Can't actually check this out anymore and the changeset looks off

Either way replaced by #82

**Pull request closed without merging**